src/source/Ice/IceAgent.c

```diff
@@ -275,22 +278,25 @@ STATUS iceAgentAddRemoteCandidate(PIceAgent pIceAgent, PCHAR pIceCandidateString
         tokenLen = (UINT32) (next - curr);
         CHK(STRNCMPI("tcp", curr, tokenLen) != 0, STATUS_ICE_CANDIDATE_STRING_IS_TCP);
 
-        if (candidateIpAddr.address[0] != 0) {
+        if (candidateIpAddr.address[0] != 0 || foundIp6) {
             CHK_STATUS(STRTOUI32(curr, curr + tokenLen, 10, &portValue));
 
             candidateIpAddr.port = htons(portValue);
-            candidateIpAddr.family = KVS_IP_FAMILY_TYPE_IPV4;
+            candidateIpAddr.family = foundIp6 ? KVS_IP_FAMILY_TYPE_IPV6 : KVS_IP_FAMILY_TYPE_IPV4;
         } else if (STRNCHR(curr, tokenLen, '.') != NULL) {
             CHK(tokenLen <= KVS_MAX_IPV4_ADDRESS_STRING_LEN, STATUS_ICE_CANDIDATE_STRING_INVALID_IP); // IPv4 is 15 characters at most
             CHK_STATUS(populateIpFromString(&candidateIpAddr, curr));
+        } else {
+            STRNCPY(ip6Buf, curr, KVS_MAX_IPV6_ADDRESS_STRING_LEN);
+            foundIp6 = inet_pton(AF_INET6, ip6Buf, candidateIpAddr.address) == 1 ? TRUE : FALSE;
         }
 
         curr = next + 1;
-        foundIpAndPort = (candidateIpAddr.port != 0) && (candidateIpAddr.address[0] != 0);
+        foundIpAndPort = (candidateIpAddr.port != 0) && ((candidateIpAddr.address[0] != 0) || foundIp6);
```

Comment:
Perhaps I am not following the logic correctly. KvsIpAddress can hold both IPv4 and IPv6 addresses. The inners of the while loop should set this param and you can check this param only. Also, consult with Felix, I am not sure why the check is against 0s. Can't the first byte of the address be 0 and the port be 0? I think an explicit 'found' boolean should be a good sentinel rather than checking for zeroed values.

Reply:
I've discussed this offline with @chehefen. He didn't write this part. But, we agreed that it makes more sense to have a boolean to check this instead. So, I'll update the PR to use a boolean instead.